pkg/cloudprovider/provider/openstack/helper.go

```diff
@@ -249,6 +251,12 @@ func getSubnet(client *gophercloud.ProviderClient, region, nameOrID string) (*os
 }
 
 func ensureKubernetesSecurityGroupExist(client *gophercloud.ProviderClient, region, name string) error {
+	// We need a mutex here because otherwise if more than one machine gets created at roughly the same time
+	// we will create two security groups and subsequently not be able anymore to identify our security group
+	// by name
+	securityGroupCreationLock.Lock()
+	defer securityGroupCreationLock.Unlock()
```

Comment:
Maybe we should remove that behavior at some time?
I dislike the fact that the controller manages more infrastructure than absolutely needed.

Reply:
I guess we could but having it there is not much of an issue either